common/src/main/java/tachyon/client/FileSystemMasterClient.java

```diff
@@ -81,14 +81,14 @@ protected void afterConnect() {
    * @return the file id for the given path
    * @throws IOException if an I/O error occurs
    */
-  public synchronized long getFileId(String path) throws IOException {
+  public synchronized long getFileId(String path) throws IOException, InvalidPathException {
     int retry = 0;
     while (!mClosed && (retry ++) <= RPC_MAX_NUM_RETRY) {
       connect();
       try {
         return mClient.getFileId(path);
       } catch (InvalidPathException e) {
-        throw new IOException(e);
+        throw e;
```

Comment:
You do, because otherwise it would be caught in the TException case, and we would retry, instead of propagating the exception.